### PR TITLE
Introduces Jest setup file that configures the React 16 Enzyme adapter

### PR DESCRIPTION
**Pull request by supernova-at**
## This PR is a:

- [ ] New feature
- [x] Enhancement/Optimization
- [x] Refactor
- [ ] Bugfix
- [ ] Test for existing code
- [ ] Documentation

## Summary

When this pull request is merged, it will alleviate developers from having to configure an Enzyme adapter at the top of every single test file.

With this change, Jest will run the the given setup file (in the `peregrine` and `venia-concept` packages) once at the start of each test file. We set up the Enzyme adapter there instead.

## Additional information

I'm definitely open to feedback on the name / location of these files.

### CODE REVIEW

File: packages/peregrine/jest.config.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
     displayName: 'Peregrine',
     setupFiles: [
         '<rootDir>/scripts/shim.js',
         '<rootDir>/scripts/fetch-mock.js'
     ],
-    browser: true,
-    testURL: 'https://localhost/',
-    coveragePathIgnorePatterns: ['scripts/*', 'node_modules', 'src/index.js'],
-    testPathIgnorePatterns: ['dist', 'node_modules']
+    setupTestFrameworkScriptFile: '<rootDir>/scripts/jest-setup.js',
```
Comment (supernova-at) [inline packages/peregrine/jest.config.js]: The addition of this property is really the only change in this file. The rest are just alphabetical order changes.

**Comment by supernova-at** [comment]
> Actually, I would consider only having one of these files. Call it jest.setup.js and put it in the root of the monorepo. See if that works.

Good thought - unfortunately, that did not work 😢 . I ran into some Babel issues and then a weird "Configured enzyme adapter did not inherit from the EnzymeAdapter base class" that the internet didn't help me solve (`venia-concept` and root `package.json` both declare `^3.7.0`).

In any event, while the duplication is not ideal, `peregrine` and `venia-concept` do both have their own `jest.config.js` which overrides the root `jest.config.js`. Perhaps in the future these newe `jest.setup.js` files introduced might diverge too.

File: packages/venia-concept/jest.config.js [hunk 2]
```diff
@@ -21,8 +20,10 @@ module.exports = {
         // improves
         '^@magento/peregrine(/*(?:.+)*)': join(__dirname, '../peregrine/src/$1')
     },
+    testPathIgnorePatterns: ['dist', 'node_modules'],
+    testURL: 'https://localhost/',
     // Have Jest use Babel to transpile Peregrine imports in tests, since
     // our cross-package tests in the monorepo should all operate on `src`
     transformIgnorePatterns: ['node_modules/(?!@magento/peregrine)'],
-    testPathIgnorePatterns: ['dist', 'node_modules']
+    setupTestFrameworkScriptFile: '<rootDir>/jest.setup.js'
```
Comment (supernova-at) [inline packages/venia-concept/jest.config.js]: More alphabetical changes here. The only real addition is the `setupTstFrameworkScriptFile` property.

**Comment by supernova-at** [comment]
> The following file(s) were not formatted with prettier. Make sure to execute npm run prettier locally prior to committing.

Is there a way to get this check to re-run? I amended a commit and had to force push, and I think that's why it didn't run again.

**Comment by zetlen** [comment]
Figured out what was unstable about the `configureHost` test (it was making hashes from filesystem paths and then testing them with a bad regex) and fixed it in this branch. Good to merge!

**Pull request merged**